src/table/utils/styling-utils.ts

```diff
@@ -42,34 +27,68 @@ export function getColor(defaultColor: string, theme: stardust.Theme, color = {}
+ * the borderLeftColor is only used for the leftmost cells, but that is added/removed by changing borderWidth
+ * the same thing applies to the borderTopColor which is only used for totals when it is at the bottom
  */
-export const getBorderColor = (isBackgroundDark: boolean): string => (isBackgroundDark ? '#F2F2F2' : '#D9D9D9');
+export const getBorderColors = (isBackgroundDark: boolean, bottomSeparatingBorder: boolean) => {
+  // TODO: proper borders for dark background
+  if (isBackgroundDark) {
+    return {
+      borderBottomColor: '#F2F2F2',
+      borderRightColor: '#F2F2F2',
+      borderLeftColor: '#F2F2F2',
+      borderTopColor: '#F2F2F2',
+    };
+  }
+
+  return {
+    borderLeftColor: '#D9D9D9',
+    borderRightColor: '#D9D9D9',
+    borderTopColor: '#808080',
+    borderBottomColor: bottomSeparatingBorder ? '#808080' : '#EBEBEB',
+  };
+};
+
+/**
+ * Get border widths for body. adds a bottom border if the rendered rows height is estimated to be greater than the container height
+ */
+const getLastRowBottomBorder = (fontSize: string | undefined, rowsLength?: number, rootElement?: HTMLElement) => {
+  if (fontSize && rowsLength && rootElement) {
+    // add line height (133%), padding and borders
+    const rowHeight = (+fontSize.slice(0, -2) * 4) / 3 + 9;
+    // multiply with number of rows plus header and totals. Compare if greater than container
+    const showBottomBorder = rowHeight * (rowsLength + 2) < rootElement.offsetHeight - PAGINATION_HEIGHT;
+    return showBottomBorder ? '1px' : '0px';
+  }
+  return '0px';
+};
 
 /**
  * Gets base styling for either header or body taking table theme settings into account
  */
 export const getBaseStyling = (
   objetName: string,
   theme: ExtendedTheme,
-  styleObj: HeaderStyling | ContentStyling | undefined
+  styleObj: HeaderStyling | ContentStyling | undefined,
+  bottomSeparatingBorder = false
 ): GeneratedStyling => {
   const fontFamily = theme.getStyle('object', `straightTable.${objetName}`, 'fontFamily');
   const color = theme.getStyle('object', `straightTable.${objetName}`, 'color');
   const fontSize = theme.getStyle('object', `straightTable.${objetName}`, 'fontSize');
 
   const baseStyle: GeneratedStyling = {
     fontFamily,
-    color: isColorSet(styleObj?.fontColor) ? getColor(StylingDefaults.FONT_COLOR, theme, styleObj?.fontColor) : color,
+    color: isPaletteColorSet(styleObj?.fontColor)
+      ? getColor(StylingDefaults.FONT_COLOR, theme, styleObj?.fontColor)
+      : color,
     fontSize: (styleObj?.fontSize && `${styleObj.fontSize}px`) || fontSize,
-    // When we do not set padding for content or header, but set font size,
-    // we need to calculate the padding based on the font size
-    padding: getPadding(styleObj),
-    borderColor: getBorderColor(theme.background.isDark),
+    padding: styleObj && 'padding' in styleObj ? styleObj?.padding : undefined,
```

Comment:
Isn't `padding: styleObj?.padding` enough?

Reply:
yeah it is a typing thing, because `styleObj: HeaderStyling | ContentStyling | undefined,`
padding only exists on contentStyling. Maybe this can be done in a cleaner way, but it was already in the `getPadding` function, so nothing new